src/Symfony/Component/Security/Http/AccessMap.php

```diff
@@ -29,7 +29,7 @@ class AccessMap implements AccessMapInterface
      * @param array                   $attributes     An array of attributes to pass to the access decision manager (like roles)
      * @param string|null             $channel        The channel to enforce (http, https, or null)
      */
-    public function add(RequestMatcherInterface $requestMatcher, array $attributes = [], $channel = null)
+    public function add(RequestMatcherInterface $requestMatcher, array $attributes = [], ?string $channel = null)
```

Comment:
CS: When `null` is the default, we omit the `?`.

Reply:
shouldn't we? https://github.com/doctrine/common/pull/805#discussion_r128762923

Reply:
As long as we need to maintain Symfony 3.4 where we don't have nullable types, this rule keeps merges from 3.4 simple. We've had similar CS rules for the old `array()` syntax when 2.8 was still maintained. Once 3.4 is in security-only mode, this rule could probably be reconsidered.

Reply:
Isn't a this PR targeting 5.0 and php 7.2?

Reply:
Yes, but the Symfony team tries to keep the code style rules consistent among branches, in order to ease merges from lower branches.

I personally always use explicit nullable types in my projects exactly for the reasons given in the comment you linked. But the Symfony project is in a different situation because of the LTS maintenance.

If you want to change Symfony's CS rules on this topic, I suggest we move that discussion to a separate issue. But you will need to convince @nicolas-grekas. 😉

Reply:
And really, this is a CS issue. On semantic level, `string $foo = null` and `?string $foo = null` are equivalent.